From 2382c38033ed7596281f82b2317f0bfbccd7d6f4 Mon Sep 17 00:00:00 2001
From: YangYonggang <b31664@freescale.com>
Date: Tue, 6 Mar 2012 17:03:42 +0800
Subject: [PATCH 248/498] ENGR00175877 System easy to hang if stop a HQ
 quality recording.

The root cause was when switching recording resolution, the frame
size was not initiated correctly.
Set frame size at the start of starting preview to fix the bug.

Signed-off-by: YangYonggang <b31664@freescale.com>
---
 mx6/libcamera/CameraHal.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/mx6/libcamera/CameraHal.cpp b/mx6/libcamera/CameraHal.cpp
index 56254bc..c9c39c7 100755
--- a/mx6/libcamera/CameraHal.cpp
+++ b/mx6/libcamera/CameraHal.cpp
@@ -1585,6 +1585,11 @@ Pic_out:
         mPPbufNum = POST_PROCESS_BUFFER_NUM;
         mTakePicFlag = false;
 
+        if(mPreviewCapturedFormat)
+                mPreviewFrameSize = mCaptureDeviceCfg.width*mCaptureDeviceCfg.height*3/2;
+            else
+                mPreviewFrameSize = mCaptureDeviceCfg.width*mCaptureDeviceCfg.height *2;
+
         if ((ret = PrepareCaptureDevices()) < 0){
             CAMERA_HAL_ERR("PrepareCaptureDevices error ");
             return ret;
@@ -1791,11 +1796,6 @@ Pic_out:
 
         //temply hard code here
         if (mTakePicFlag == 0){
-            if(V4L2_PIX_FMT_NV12)
-                mPreviewFrameSize = mCaptureDeviceCfg.width*mCaptureDeviceCfg.height*3/2;
-            else 
-                mPreviewFrameSize = mCaptureDeviceCfg.width*mCaptureDeviceCfg.height *2;
-
             if(mPreviewMemory != NULL) {
                 mPreviewMemory->release(mPreviewMemory);
             }
-- 
1.8.0

